### PR TITLE
Fix stringop-overflow error when building with gcc 12.x

### LINKED ISSUE #1577: stringop-overflow error when building with gcc 12.x
I upgraded my laptop to Fedora 36 yesterday and evidently inherited a newer `gcc`:

```
$ gcc --version
gcc (GCC) 12.1.1 20220507 (Red Hat 12.1.1-1)
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

This leads to the following error during compilation:

```
[ 50%] Building C object src/crypto/CMakeFiles/cncrypto.dir/oaes_lib.c.o
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c: In function ‘oaes_encrypt’:
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1276:52: error: writing 1 byte into a region of size 0 [-Werror=stringop-overflow=]
 1276 |                         _block[ _block_size + _j ] = _j + 1;
      |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/home/ianmacd/src/oxen-core/src/crypto/oaes_lib.c:1269:25: note: at offset 16 into destination object ‘_block’ of size 16
 1269 |                 uint8_t _block[OAES_BLOCK_SIZE];
      |                         ^~~~~~
cc1: all warnings being treated as errors
make[2]: *** [src/crypto/CMakeFiles/cncrypto.dir/build.make:286: src/crypto/CMakeFiles/cncrypto.dir/oaes_lib.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1475: src/crypto/CMakeFiles/cncrypto.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

I will post a patch after filing this issue.

### PR DESCRIPTION
**Pull request by ianmacd**
Fixes #1577.

### CODE REVIEW

File: src/crypto/oaes_lib.c [hunk 1]
```diff
@@ -1273,7 +1273,8 @@ OAES_RET oaes_encrypt( OAES_CTX * ctx,
 		
 		// insert pad
 		for( _j = 0; _j < OAES_BLOCK_SIZE - _block_size; _j++ )
-			_block[ _block_size + _j ] = _j + 1;
+			uint8_t offset = _j;
```
Comment (majestrate) [inline src/crypto/oaes_lib.c:1276]: needs to be in a curly brace block as this for loop without one will be just the first line.

also iirc, there actually may be a bad side effect in the old code that is required here.

Comment (majestrate) [reply src/crypto/oaes_lib.c:1276]: also, i dont think this has the same logic as before.

**Comment by majestrate** [comment]
can we not change all the whitespaces here as well?

**Comment by ianmacd** [comment]
This is perhaps the worst PR I've ever submitted.

I hang my head in shame.

I will start again from scratch.

**Pull request closed without merging**